src/server/ua_services_discovery.c

```diff
@@ -361,7 +822,7 @@ void Service_RegisterServer(UA_Server *server, UA_Session *session,
  */
 void UA_Discovery_cleanupTimedOut(UA_Server *server, UA_DateTime nowMonotonic) {
 
-    UA_DateTime timedOut = nowMonotonic;
+    UA_DateTime timedOut = UA_DateTime_now();
```

Comment:
We use the monotonic clock for timeouts to avoid closing connections when the system time is updated.
There have been bug reports in this regard.